x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/GetEngineAction.java

```diff
@@ -0,0 +1,146 @@
+
+package org.elasticsearch.xpack.entsearch.engine.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.entsearch.engine.Engine;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+import static org.elasticsearch.xpack.entsearch.engine.EngineIndexService.ENGINE_ALIAS_NAME;
+
+public class GetEngineAction extends ActionType<GetEngineAction.Response> {
+
+    public static final GetEngineAction INSTANCE = new GetEngineAction();
+    public static final String NAME = "indices:admin/engine/get";
+
+    private GetEngineAction() {
+        super(NAME, GetEngineAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest implements IndicesRequest.Replaceable {
+
+        public static final IndicesOptions DEFAULT_INDICES_OPTIONS = IndicesOptions.strictExpandOpen();
+
+        private String[] names = new String[] { ENGINE_ALIAS_NAME };
+        private final IndicesOptions indicesOptions = DEFAULT_INDICES_OPTIONS;
+
+        private final String engineId;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.engineId = in.readString();
+        }
+
+        public Request(String engineId) {
+            this.engineId = engineId;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            if (engineId == null || engineId.isEmpty()) {
+                validationException = addValidationError("engineId missing", validationException);
+            }
+
+            return validationException;
+        }
+
+        public String getEngineId() {
+            return engineId;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(engineId);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request request = (Request) o;
+            return Objects.equals(engineId, request.engineId);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(engineId);
+        }
+
+        @Override
+        public IndicesRequest indices(String... indices) {
+            this.names = indices;
```

Comment:
@jimczi , I guess this method won't be used since we won't be using wildcards or index resolution from the indices() method, and that method will always return the Engine alias.

Could you please explain what can trigger this method to be invoked? What should we do in case it is, update the Engine alias name? 🤔

Reply:
> Could you please explain what can trigger this method to be invoked?
It should happen only if there are index patterns in the list of indices. Since we disallow index patterns, I like @kderusso 's idea to throw an `IllegalStateException` in this method if that's possible.

Reply:
Does it make sense for this class to implement just `IndicesRequest` instead of `IndicesRequest.Replaceable` in that case?

Reply:
I'm going to punt on this change for now, simply because making `names` final and throwing made the tests fail. We can revisit it another time though.